lintro/ai/providers/cursor.py

```
@@ -187,130 +272,28 @@ def complete(
-                    "Run 'agent login' or set CURSOR_API_KEY.",
-                )
-            raise AIProviderError(
-                f"Cursor CLI exited with code {result.returncode}: {stderr}",
+            self._cli.check_exit_code(
+                result,
+                auth_patterns=("authentication required", "login"),
+                auth_hint="Run 'agent login' or set CURSOR_API_KEY.",
             )
+            response, session_id = self._cli.parse_stdout(result.stdout)
+        except AIAuthenticationError:
+            raise
```

Ruff (0.15.20)

[warning] Remove exception handler; error is immediately re-raised

(TRY203)